pennylane/transforms/mitigate.py

```diff
@@ -525,6 +525,11 @@ def circuit(w1, w2):
 
     def processing_fn(results):
         """Maps from input tape executions to an error-mitigated estimate"""
+
+        # content of `results` must be modified in this post-processing function
+        if isinstance(results, tuple):
+            results = list(results)
+
```

Comment:
```suggestion

        # content of `results` must be modified in this post-processing function
        results = list(results)

```

This would also essentially "copy" a list, which would potentially prepare it for mutation.